File: 2016_05_01/PICircularProgressView/PICircularProgressView/AppDelegate+TestVC.m
//
//  AppDelegate+TestVC.m
//  UserInfoLearn1
//
//  Created by xiaoyulong on 17/3/20.
//  Copyright © 2017年 xiaoyulong. All rights reserved.
//

#import "AppDelegate+TestVC.h"


@implementation AppDelegate (TestVC)

- (void)setupTestRootVC
{
    
    NSMutableArray *array = [[NSMutableArray alloc] init];
    

    
    
    // MARK: - 2016年面试遇到的问题
    [array addObject:@"TestVC1"];
    
    

    // MARK: - 圆环
    /**
     pod 'PICircularProgressView', '~> 0.1.1'
     - Homepage: https://github.com/DerPipo/PICircularProgressView
     */
    [array addObject:@"RootViewController"];
    
    
    [self makeControllerWith:array];
}

- (void)makeControllerWith:(NSArray *)array
{
//    NSString *clsName = array[0];
    
    
    NSMutableArray *mArray = [[NSMutableArray alloc] init];
    for (NSInteger i = array.count-1; i >= 0; --i)
    {
        [mArray addObject:array[i]];
    }
    NSString *clsName = mArray[1];
    
    
    
    Class class = NSClassFromString(clsName);
    
    UIViewController *rootVC = [[class alloc] init];
    UINavigationController *nav = [[UINavigationController alloc] initWithRootViewController:rootVC];
    rootVC.navigationItem.title = NSStringFromClass(rootVC.classForCoder);
    rootVC.edgesForExtendedLayout = UIRectEdgeNone;
    rootVC.automaticallyAdjustsScrollViewInsets = NO;
    rootVC.extendedLayoutIncludesOpaqueBars = NO;
    self.window.rootViewController = nav;
    
}


@end

File: 2017_06_01/FBRetainCycleDetectorLearn2/FBRetainCycleDetectorLearn2/AppDelegate+TestVC.swift
//
//  AppDelegate+TestVC.swift
//  FBRetainCycleDetectorLearn2
//
//  Created by xiaoyulong on 2017/4/25.
//  Copyright © 2017年 xiaoyulong. All rights reserved.
//

import Foundation
import UIKit

extension AppDelegate {

    func setupTestRootVC() {
        let array = [
        
            
            "TestVC1",
            
            
//            http://www.alonemonkey.com/2016/05/15/fbretaincycledetector-analyse/
           
            // MARK: - https://github.com/facebook/FBRetainCycleDetector
            "RootViewController"
        
        ]
        
        makeController(arrary: array)
    }
    
    
    func makeController(arrary: [String]) {
//        let clsName = arrary[0]
        
        let clsName = arrary.reversed()[1]
        
        guard let cls = NSClassFromString(Bundle.main.namespace + "." + clsName) as? UIViewController.Type else {
            return
        }
        
        let vc = cls.init()
        vc.navigationItem.title = clsName
        let nav = UINavigationController.init(rootViewController: vc)
        self.window?.rootViewController = nav
    }
}
